dropwizard-core/src/main/java/io/dropwizard/server/AdminServletFactory.java

```diff
@@ -0,0 +1,235 @@
+import com.codahale.metrics.servlets.HealthCheckServlet;
+import com.codahale.metrics.servlets.MetricsServlet;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.jetty.MutableServletContextHandler;
+import org.eclipse.jetty.server.handler.ContextHandler.Context;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+/**
+ * A factory for configuring the administrative servlet (and its child servlets) for the environment.
+ * <p/>
+ * <b>Configuration Parameters:</b>
+ * <table>
+ *     <tr>
+ *         <td>Name</td>
+ *         <td>Default</td>
+ *         <td>Description</td>
+ *     </tr>
+ *     <tr>
+ *         <td>metricsEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether metrics information will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>metricsUri</td>
+ *         <td>/metrics</td>
+ *         <td>The relative URI for metrics retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>pingEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether a {@code ping} liveness function will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>pingUri</td>
+ *         <td>/ping</td>
+ *         <td>The relative URI for liveness response retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>threadsEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether thread dump information will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>threadsUri</td>
+ *         <td>/threads</td>
+ *         <td>The relative URI for thread dump retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>healthcheckEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether health check information will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>healthcheckUri</td>
+ *         <td>/healthcheck</td>
+ *         <td>The relative URI for health check retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>cpuProfileEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether CPU profiling results will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>cpuProfileUri</td>
+ *         <td>/pprof</td>
+ *         <td>The relative URI for CPU profiling retrieval.</td>
+ *     </tr>
+ * </table>
+ *
+ * @since 2.0
+ */
+public class AdminServletFactory {
```

Comment:
Could you please add a test for this class?

Reply:
Yes, as soon as I can get back to this; is there an example I could use for reference?

Reply:
I've added tests for the factory based on `GzipHandlerFactoryTest`